test/auth/auth_test.go

```diff
@@ -259,3 +259,29 @@ func getAccessToken(t *testing.T, weaviateUrl, user, pw string) (string, string)
 	require.Nil(t, err)
 	return tokens.AccessToken, tokens.RefreshToken
 }
+
+func TestApiKey(t *testing.T) {
+	t.Run(t.Name(), func(t *testing.T) {
+		url := fmt.Sprintf("127.0.0.1:%v", testsuit.WCSPort)
+		headers := map[string]string{}
+		cfg, err := weaviate.NewConfig(url, "http", auth.ApiKeys{ApiKey: "my-secret-key"}, headers)
+		assert.Nil(t, err)
+
+		client := weaviate.New(*cfg)
+		AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+		assert.Nil(t, AuthErr)
+	})
+}
+
+func TestWrongApiKey(t *testing.T) {
+	t.Run(t.Name(), func(t *testing.T) {
+		url := fmt.Sprintf("127.0.0.1:%v", testsuit.WCSPort)
+		headers := map[string]string{}
+		cfg, err := weaviate.NewConfig(url, "http", auth.ApiKeys{ApiKey: "wrong_key"}, headers)
+		assert.Nil(t, err)
+
+		client := weaviate.New(*cfg)
+		AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+		assert.NotNil(t, AuthErr)
+	})
+}
```

Comment:
could you please add a mock_test with a weavaite mock where OIDC is not enabled? Here is the relevant python test: https://github.com/weaviate/weaviate-python-client/blob/035d45e0159c75e41c779302f2b726c67c07cc12/mock_tests/test_auth.py#L208